tests/std/tests/GH_001596_adl_proof_algorithms/test.compile.pass.cpp

```diff
@@ -263,6 +263,48 @@ void test_algorithms() {
     // (void) std::shift_right(varr, varr, 0); // requires Cpp17ValueSwappable
 #endif // _HAS_CXX20
 
+    int iarr3[1]{};
+    validator varr3[1]{};
+
+    (void) std::merge(varr, varr, varr2, varr2, varr3);
+    (void) std::merge(iarr, iarr, iarr2, iarr2, iarr3, validating_less{});
+
+    // std::inplace_merge(varr, varr, varr); // requires Cpp17ValueSwappable
+    std::inplace_merge(iarr, iarr, iarr, validating_less{});
+
+    (void) std::includes(varr, varr, varr, varr);
+    (void) std::includes(iarr, iarr, iarr, iarr, validating_less{});
+
+    (void) std::set_union(varr, varr, varr, varr, varr3);
+    (void) std::set_union(iarr, iarr, iarr, iarr, iarr3, validating_less{});
+
+    (void) std::set_intersection(varr, varr, varr, varr, varr3);
+    (void) std::set_intersection(iarr, iarr, iarr, iarr, iarr3, validating_less{});
+
+    (void) std::set_difference(varr, varr, varr, varr, varr3);
+    (void) std::set_difference(iarr, iarr, iarr, iarr, iarr3, validating_less{});
+
+    (void) std::set_symmetric_difference(varr, varr, varr, varr, varr3);
+    (void) std::set_symmetric_difference(iarr, iarr, iarr, iarr, iarr3, validating_less{});
+
+    std::push_heap(varr3, varr3 + 1); // requires Cpp17ValueSwappable

```

Comment:
It seems that MSVC STL's `std::push_heap`, `std::pop_heap`, `std::make_heap`, and `std::sort_heap` don't require _Cpp17ValueSwappable_ (i.e. don't use ADL-found `swap`) currently. Should we keep these test cases?

Reply:
Nor do libstdc++'s and libc++'s ([Godbolt link](https://godbolt.org/z/1edq69G7x)).

Reply:
I believe we should. The fact that we don't swap today doesn't mean we never will; these test cases are useful to prevent regression.

Reply:
I'm fine with this for now.

Reply:
@CaseyCarter If I understand this correctly, if we started ADL-swapping in `meow_heap` in the future, these test cases would inherently fail to compile, because ADL-swapping is incompatible with the Incomplete scenario. I'm fine with that because we'd just comment them out like the other swapping test cases.